File: Assignments/Assignment_66/program2.java
// write a program which accept one number from user and position from user and off that bit.Return modified number.

import java.util.Scanner;

class Bitwise
{
    public int OffBit(int iNo, int iPos)
    {
        int iMask = 0x1, iResult = 0;

        iMask = iMask << (iPos - 1);

        iResult = iNo & (~iMask);

        return iResult;
    }
}

class program2
{
    public static void main(String A[])
    { 
        int iValue = 0, iPos = 0;
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Number : ");
        iValue = sobj.nextInt();

        System.out.println("Enter Position : ");
        iPos = sobj.nextInt();

        Bitwise bobj = new Bitwise();
         iRet = bobj.OffBit(iValue,iPos);

        System.out.println("Modified number is : "+iRet);
        
    }
}

File: Assignments/Assignment_71/program2.c
// Write a recursive program which Accept Number from user and return largest digit.
// Input  : 87983
// Output : 9

#include<stdio.h>

int Max(int iNo)
{
    static int iMax = 0;
    static int iDigit = 0;

    if(iNo != 0)
    {
        iDigit = iNo % 10;
        
        if(iMax < iDigit)
        {
            iMax = iDigit;
        }

        iNo = iNo / 10;

        Max(iNo);
    }

    return iMax;
}

int main()
{
    int iValue = 0, iRet = 0;

    printf("Enter Number : \n");
    scanf("%d",&iValue);

    iRet = Max(iValue);

    printf("%d",iRet);

    return 0;
}
